From: Michal Kubecek <mkubecek@suse.cz>
Date: Mon, 17 Oct 2022 15:15:55 +0200
Subject: scsi: mpi3mr: add explicit dependency on CONFIG_SCSI_SAS_ATTRS
Patch-mainline: Submitted - 2022-10-17 - 20221017140533.CC46F6043B@lion.mk-sys.cz
References: none

Starting with commit 42fc9fee116f ("scsi: mpi3mr: Add helper functions to
manage device's port"), kernel configured with CONFIG_SCSI_MPI3MR=m and
CONFIG_SCSI_SAS_ATTRS=n fails to build because modpost cannot find symbols
used in mpi3mr_transport.c:

  ERROR: modpost: "sas_port_alloc_num" [drivers/scsi/mpi3mr/mpi3mr.ko] undefined!
  ERROR: modpost: "sas_remove_host" [drivers/scsi/mpi3mr/mpi3mr.ko] undefined!
  ERROR: modpost: "sas_phy_alloc" [drivers/scsi/mpi3mr/mpi3mr.ko] undefined!
  ERROR: modpost: "sas_phy_free" [drivers/scsi/mpi3mr/mpi3mr.ko] undefined!
  ...

Add an explicit dependency of CONFIG_SCSI_MPI3MR on CONFIG_SCSI_SAS_ATTRS
to prevent inconsistent configs.

Fixes: 42fc9fee116f ("scsi: mpi3mr: Add helper functions to manage device's port")
Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 drivers/scsi/mpi3mr/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/scsi/mpi3mr/Kconfig
+++ b/drivers/scsi/mpi3mr/Kconfig
@@ -2,7 +2,7 @@
 
 config SCSI_MPI3MR
 	tristate "Broadcom MPI3 Storage Controller Device Driver"
-	depends on PCI && SCSI
+	depends on PCI && SCSI && SCSI_SAS_ATTRS
 	select BLK_DEV_BSGLIB
 	help
 	MPI3 based Storage & RAID Controllers Driver.
